src/box/index.h

```diff
@@ -327,6 +337,12 @@ struct iterator {
 	 * Don't access directly, use iterator_space().
 	 */
 	struct space *space;
+	/**
+	 * Pointer to a buffer which was allocated for start position with
+	 * malloc. Will be freed on iterator_delete.
+	 * Needed for box_index_iterator_after.
+	 */
+	char *blessed_position_buf;
```

Comment:
Maybe, store it in the iterator implementation struct?

Rationale:
 - Vinyl already copies the position internally.
 - The iterator may choose to free the position right after the first iteration, because it's no longer needed (Vinyl already does that).
 - The iterator may start the iteration on open and not store the position at all.

Reply:
Good idea, but this brings up an API question - how should we pass this buffer?

Reply:
- A separate method looks weird.
- A new argument in `create_iterator` method looks ugly. Even more, if we will use runtime_alloc instead of malloc, we will need to store size of buffer - it is even more ugly.
Maybe we should keep it simple, especially because 99% of `pairs` usage is `for` loop.

Reply:
Agree - passing the position size to the `create_iterator` callback would be a problem. Let's leave it as is for now.